### DISCUSSION TITLE
Can we have a way to run GPRESULT

### DISCUSSION THREAD

**Discussion by we-sell-bags** [glpi-project/glpi-agent#280] [Ideas]
on windows, would it be possible to have a way to run "GPREPORT" and pass back the html file.

when applying GPO to computers it is useful to know what worked & what did not and on what machine.
currently getting GPREPORTS from windows machines is a complete pain , ESP. if remote desktop is not turned on.

**Comment by g-bougard** [comment]
Hi @we-sell-bags 

maybe Deploy task can help. Deploy task can be used to run any kind of payload, not only to install a software. You can create a script which will upload the report yo a shared place or a dedicated service.

Or we need a new feature in GLPI:
1. the availability to upload a computer document to GLPI server so it is attached to the computer asset in a dedicated frame,
2. then GLPI agent could be modified to automatically upload discovered files and "GPREPORT" could be a good candidate.

Also did you investigate to use an IT automation tool like Ansible ?

**Comment by we-sell-bags** [comment]
Yep... 
I know about such tools as 'Ansible'
I'm trying to keep the tools and what we have on users systems as simple as possible.
Every  tool  gets the Auditors on a paper trail i have to provide.
I'll take a look at  deploy tasks..
